### ISSUE TITLE
gradients cannot propagate correctly on the cloned lora model

### ISSUE THREAD

**Issue by luguangyao** [huggingface/peft#634]
### System Info

Hello, recently I use learn2learn package for meta learning task. I use lora as the original model, and use learn2learn package to get a replica of lora. But when I used the replica to calculate the gradient, I found that the gradient did not propagate on the replica, but rather on the original model. I spent a few days trying to find the cause and finally found that the problem was on line 152 of peft tuner lora. py:
`self.forward = self.model.forward`
This line of code causes the forward function of the replica to point to the forward function of the original model, causing the data to run on the original model.
After I add this code, the gradients propagate normally:
`learner.module.base_model.forward=learner.module.base_model.model.forward`

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [X] My own task or dataset (give details below)

### Reproduction

```
loss_fn=torch.nn.CrossEntropyLoss()
# model is a lora model for SequenceClassification, learner is a replica
model = l2l.algorithms.MAML(
                    lora_model,
                    lr=args.fast_lr,
                    allow_nograd=True)
learner=model.clone()

# gradients did not propagate normally on the replicate
loss=loss_fn(learner(**tx).logits, ty)
diff_param=[p for p in learner.parameters() if p.requires_grad]
grad=torch.grad.autograd(loss, diff_papram, retain_graph=True, create_graph=True, allow_unused=True)
print('grad is' ,grad)
# grad is [None, None, ... , None]

# gradients propagated on the original model
diff_param=[p for p in model.parameters() if p.requires_grad]
grad=torch.grad.autograd(loss, diff_papram, retain_graph=True, create_graph=True, allow_unused=True)
print('grad is' ,grad, sep='\n')
# grad is
# (tensor([[0., 0., 0.,  ..., 0., 0., 0.],
#        [0., 0., 0.,  ..., 0., 0., 0.],
#        [0., 0., 0.,  ..., 0., 0., 0.],
#         ...,
#        [0., 0., 0.,  ..., 0., 0., 0.],
#        ...,

# gradients propagate normally on the replicate
# add the code below to fix the problem
learner.module.base_model.forward=learner.module.base_model.model.forward
loss=loss_fn(learner(**tx).logits, ty)
diff_param=[p for p in learner.parameters() if p.requires_grad]
grad=torch.grad.autograd(loss, diff_papram, retain_graph=True, create_graph=True, allow_unused=True)
print('grad is' ,grad)
print('grad is' ,grad, sep='\n')
# grad is
# (tensor([[0., 0., 0.,  ..., 0., 0., 0.],
#        [0., 0., 0.,  ..., 0., 0., 0.],
#        [0., 0., 0.,  ..., 0., 0., 0.],
#         ...,
#        [0., 0., 0.,  ..., 0., 0., 0.],
#        ...,
```

### Expected behavior

When performing meta learning or model parallelism, could you please fix this issue？
